internal/provider/resource_edge_cluster.go

```diff
@@ -0,0 +1,250 @@
+				Description: "One among: XLARGE, LARGE, MEDIUM, SMALL",
+			},
+			"high_availability": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "One among: ACTIVE_ACTIVE, ACTIVE_STANDBY",
+			},
+			"mtu": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "Maximum transmission unit size for the cluster",
+			},
+			"asn": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "ASN for the cluster",
+			},
+			"skip_tep_routability_check": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Set to true to bypass normal ICMP-based check of Edge TEP / host TEP routability (default is false, meaning do check)",
+			},
+			"tier1_unhosted": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Select whether Tier-1 being created per this spec is hosted on the new Edge cluster or not (default value is false, meaning hosted)",
+			},
+			"internal_transit_subnets": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Subnet addresses in CIDR notation that are used to assign addresses to logical links connecting service routers and distributed routers",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"transit_subnets": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Transit subnet addresses in CIDR notation that are used to assign addresses to logical links connecting Tier-0 and Tier-1s",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"edge_node": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "The nodes in the edge cluster",
+				Elem:        edge_cluster.EdgeNodeSchema(),
+			},
+		},
+	}
+}
+
+func resourceEdgeClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*api_client.SddcManagerClient).ApiClient
+
+	params := &nsxt_edge_clusters.CreateEdgeParams{
+		EdgeCreationSpec: edge_cluster.GetEdgeClusterCreationSpec(d),
+		Context:          ctx,
+	}
+
+	_, task, err := c.NSXTEdgeClusters.CreateEdge(params)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	tflog.Info(ctx, "Edge cluster creation has started.")
+	err = meta.(*api_client.SddcManagerClient).WaitForTaskComplete(ctx, task.Payload.ID, false)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var clusters *nsxt_edge_clusters.GetEdgeClustersOK
+	clusters, err = c.NSXTEdgeClusters.GetEdgeClusters(nil)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	for _, cluster := range clusters.Payload.Elements {
+		if cluster.Name == d.Get("name") {
+			d.SetId(cluster.ID)
+			tflog.Info(ctx, "Edge cluster created successfully.")
+			return nil
+		}
+	}
+
+	return diag.Errorf("Edge cluster creation failed - cluster not found in inventory")
+}
+
+func resourceEdgeClusterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
There are no computed properties?

Reply:
No, the only computed value is the identifier and that is assigned to the resource.
There are no properties on the cluster that can be modified post-creation (aside from the set of nodes).